### PR TITLE
berniem module 2

### PR DESCRIPTION
**Pull request by httpstergeek**
I tried to use define as listed in your final example but couldn't get it to work.

### CODE REVIEW

File: app/js/enhancedDate.js [hunk 1]
```diff
@@ -1,5 +1,99 @@
...
+ */
 
+var enhancedDate = (function iffe() {
+  'use strict';
+  function now() {
+    return new Date();
+  }
+
+  var dateObjects = {
+    dateTime: now(),
```
Comment (jamdon) [inline app/js/enhancedDate.js]: By invoking this here the time is set as soon as the module code is loaded. What may be more flexible would be to only set it once one of the methods on the enhancedDate instance is called.

File: app/js/enhancedDate.js [hunk 2]
```diff
@@ -1,5 +1,99 @@
...
+      }
+      else if (e === undefined) {
+        dateObjects.dateTime = new Date();
+      }
+      else {
+        console.warn('argument is not instanceof Date');
+      }
+      return dateObjects.dateTime;
+    },
+    getDate: function getDate(e) {
```
Comment (jamdon) [inline app/js/enhancedDate.js:52]: If nothing is passed to getDate, it should return the date in milliseconds past epoch.

File: app/js/enhancedDate.js [hunk 3]
```diff
@@ -1,5 +1,99 @@
...
+      }
+      else {
+        return false;
+      }
+    },
+    isToday: function isToday() {
+      var year = now().getFullYear();
+      var month = now().getMonth();
+      var day = now().getDate();
+      var nextDay = new Date(year, month, day).getTime() + 86400000;
```
Comment (jamdon) [inline app/js/enhancedDate.js]: Values like this are better declared as a constant elsewhere.

Comment (httpstergeek) [inline app/js/enhancedDate.js]: just for clarification.  Are you saying that line 76 the value of day in milliseconds should be stored else where?

Comment (jamdon) [inline app/js/enhancedDate.js]: Sort of. What I mean is declare a variable as a constant somewhere and use
that variable there - makes the code more self-documenting:

var MILLISECONDS_IN_DAY = 86400000;

var nextDay = new Date(year, month, day).getTime() + MILLISECONDS_IN_DAY;

In general, avoid magic numbers in code so the code's intent is immediately
obvious.

**Comment by jamdon** [comment]
Nice work. And the define( ) example was only to demonstrate what some of the module libraries might look like when you define your own module using them. There isn't currently any native implementation of a define method. :+1:

**Pull request closed without merging**